spacewalk/setup/share/embedded_diskspace_check.py

```diff
@@ -195,7 +192,6 @@ def main(needsDict=None):
 if __name__ == "__main__":
     needsDict = {}
     if sys.argv:
-        for i, dir in enumerate(sys.argv[1::2]):
+        for i, dir in enumerate(sys.argv[1::2]):  # pylint: disable=redefined-builtin
```

Comment:
I'd throw this away and just use [`argparse`](https://docs.python.org/2/howto/argparse.html) instead. It is easier and more neat.

Reply:
In this particular case, I think it's not worth to migrate this to `argparse`. This file is a simple script which is called along with a string containing pairs of paths/sizes. Then the script process this data to produces the corresponding output.

IMHO `argparse` is meant for more complex and structured argument handling which is not needed at all here.

Reply:
How it is complex? It is not complex. The current code is just nuts. All it does, is turning even array into a dictionary without _any_ error checks:

```
['program_name', 'a', 123, 'c', 234]
```
into this:
```
('a', 123)
('c', 234)
```

How this is any good?

Instead, usage of argparse gives you a better idea what this tool supposed to do. This program is called from Perl `System.pm`, so we should preserve the command line:

```python
import argparse

p = argparse.ArgumentParser()
p.add_argument('DEFAULT_NEEDS', nargs='+',
               help=('Set default map of failed needs: [directory] [size] ... '
                     'Example: /var/lig/pgsql/data 12288'))
args = p.parse_args()

if len(args.DEFAULT_NEEDS) % 2:
    p.print_help()
else:
    needs_dict = {}
    for directory, size in zip(args.DEFAULT_NEEDS[0::2], args.DEFAULT_NEEDS[1::2]):
        try:
            needs_dict[directory] = int(size)
        except ValueError as err:
            p.error(err)
     sys.exit(main(needs_dict) or 0)
```

The code above fixes the following bugs and coding errors:

- This _requires_ to put pairs of integer/directory, all even pairs. If you put odd amount of arguments, this will crash.
- The `dir` is preserved Python word and we reusing it. This is not a good tone.
- PyLint is right, let's not silence this.
- Get rid of complete nonsense coded here.
- @brejoc will be reading easier.

Reply:
I agree with you that this code could be refactored in a better, elegant and more efficient way, but not only this particular line but the entire file. But the scope of this PR is porting the code to be Python 3 compatible, it's not about refactoring ugly code which is already compatible.

I simply added the `pylint: disable=redefined-builtin` tag there because it was needed in order to get green pylint checks during package building (which are now disabled). So this is not even needed now and could be removed.

IMHO these type of changes are out of the scope of this PR. Of course it would be very nice to improve the code base in general, but I think they should not be required for PR approval.

Reply:
I'm afraid I have to agree with @meaksh here. If we mix two topics, things might get rejected because of the other topic that should not be addressed initially. So the refactoring should be done in a second PR. :peace_symbol:

Reply:
@meaksh if you swear on Avocado we will refactor that soon, let it go then. :-)

Reply:
As discussed, I've created a card to do this refactor on a separated PR: https://github.com/SUSE/salt-board/issues/185

Reply:
But @meaksh 's case this should be a mango! :laughing:

Reply:
I'm resolving this conversation since spin-off card has been created.